Registry/Platform/Mac/AssetProcessorPlatformConfig.setreg

```diff
@@ -0,0 +1,16 @@
+{
+    "Amazon": {
+        "AssetProcessor": {
+            "Settings": {
+                "Exclude Terrain Materials": {
+                    // The terrain shader doesn't work on mac due to unbounded arrays, so disable all mac assets
+                    // in the terrain gem to prevent dependencies from failing.
+                    "pattern": "(^|.+/)Gems/Terrain/Assets/Materials(/.*)?$"
+                },
+                "Exclude Terrain Shaders": {
+                    "pattern": "(^|.+/)Gems/Terrain/Assets/Shaders(/.*)?$"
+                }
```

Comment:
This exclude rule will not work.

Exclusion rules are run relative to the Scan Folders within the AssetProcessor.

The ScanFolder for the Terrain Gem would be the `<engine-root>/Gems/Terrain/Assets`, directory.
Therefore any exclusion rule would need to be relative to that path.

My suggestion is instead of using an exclusion to prevent the processing of assets, you can remove the Scan Folder for the Terrain Gem.

Reply:
Do we have a way to remove a scan folder?  I was under the impression that there's some hard code in there that auto adds 'assets' if it exists in a gem thats enabled?

Reply:
The code does auto add the "Assets" folder to the list of Scan Folders, but the Scan Folder configuration does support an "exclude" field which can be supplied with the list of tags to exclude.

Reply:
I'm not sure if `AddGemScanFolders()` looks for the exclude field when adding the individual gem asset folders.

Reply:
It does not, but having a custom `ScanFolder gemassets-<UUID>` entry in a setreg that has the same portable key as the one generated for the Terrain Gem inside of `AddGemScanFolders()` can be used to override the ScanFolder settings via the [ReadRecognizersFromSettingsRegistry](https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/utilities/PlatformConfiguration.cpp#L1012) function.

That being said, the Uuid for the name "Terrain" would need to be gather by debugging the code and putting a breakpoint inside of `AddGemScanFolders` to discover it in the first place

Reply:
Looking at all the approaches we have available to us, the best one is the exclusion rule approach, but it would need to be very targeted to the complete relative paths of files to exclude.

Reply:
One other way might be to move the assets to subfolder named something like "TerrainPlatformExcludes" and then use that as an exclude pattern.